pom.xml

```diff
@@ -764,5 +790,57 @@ limitations under the License.
         </plugins>
       </build>
     </profile>
+
+    <profile>
+      <id>java11+</id>
+      <activation>
+        <jdk>[11,)</jdk>
+      </activation>
+
+      <properties>
+        <maven.compiler.release>${javaVersion}</maven.compiler.release>
+        <!-- configuration for checkstyle to use with spotless -->
+        <checkstyle.spotless.config>config/maven_checks_nocodestyle.xml</checkstyle.spotless.config>
+      </properties>
+
+      <build>
+        <!--- newer versions of plugins requires JDK 11 -->
+        <plugins>
+          <plugin>
+            <groupId>com.diffplug.spotless</groupId>
+            <artifactId>spotless-maven-plugin</artifactId>
+          </plugin>
+          <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-checkstyle-plugin</artifactId>
+            <configuration>
+              <configLocation>${checkstyle.spotless.config}</configLocation>
+            </configuration>
+          </plugin>
+        </plugins>
+      </build>
+    </profile>
+    <profile>
+      <id>format-check</id>
```

Comment:
IMHO additional profiles are not needed ... I simply execute ` mvn spotless:apply`